### PR TITLE
ref(tx-clusterer): Instrument unclustered projects

### PR DESCRIPTION
**Pull request by iker-barriocanal**
All projects should be clustered, including any time limit issues. Adding a metric about the number of unclustered projects provides visibility into any potential issues.

### CODE REVIEW

File: src/sentry/ingest/transaction_clusterer/tasks.py [hunk 1]
```diff
@@ -47,28 +47,36 @@ def spawn_clusterers(**kwargs: Any) -> None:
     queue="transactions.name_clusterer",
     default_retry_delay=5,  # copied from release monitor
     max_retries=5,  # copied from release monitor
+    soft_time_limit=5,
+    time_limit=7,  # extra 2s to emit metrics
```
Comment (iker-barriocanal) [inline src/sentry/ingest/transaction_clusterer/tasks.py:51]: Tasks take ~2.5s on average, with max peaks of ~3.5s. A 5s limit is rather conservative to not trigger any thresholds in the current state.

File: src/sentry/ingest/transaction_clusterer/tasks.py [hunk 2]
```diff
@@ -47,28 +47,36 @@ def spawn_clusterers(**kwargs: Any) -> None:
...
-            # Clear transaction names to prevent the set from picking up
-            # noise over a long time range.
-            redis.clear_transaction_names(project)
+                # Clear transaction names to prevent the set from picking up
+                # noise over a long time range.
+                redis.clear_transaction_names(project)
+            num_clustered += 1
+    finally:
+        unclustered = len(projects) - num_clustered
+        metrics.incr("txcluster.cluster_projects.unclustered", amount=unclustered, sample_rate=1.0)
```
Comment (jjbayer) [inline src/sentry/ingest/transaction_clusterer/tasks.py:82]:
```suggestion
        if unclustered:
            metrics.incr("txcluster.cluster_projects.unclustered", amount=unclustered, sample_rate=1.0)
```

File: src/sentry/ingest/transaction_clusterer/tasks.py (hunk 1 above)
Comment (jjbayer) [reply src/sentry/ingest/transaction_clusterer/tasks.py:51]: 1. Most other Celery tasks have a much larger timeout, I would feel more comfortable if we took ~4x the average, i.e. 10 seconds.
2. Can we make these timeouts depend on `PROJECTS_PER_TASK`? E.g.
```suggestion
    soft_time_limit=0.1 * PROJECTS_PER_TASK, # 100 ms per project
    time_limit=0.1 * PROJECTS_PER_TASK + 2,  # extra 2s to emit metrics
```

**Review by jjbayer — CHANGES_REQUESTED** [review]

File: src/sentry/ingest/transaction_clusterer/tasks.py [hunk 3]
```diff
@@ -47,28 +52,39 @@ def spawn_clusterers(**kwargs: Any) -> None:
...
-                clusterer.add_input(tx_names)
-                new_rules = clusterer.get_rules()
+    num_clustered = 0
+    try:
+        for project in projects:
+            # NOTE: The probability that the feature flag is True is high, because
+            # we know at this point that a redis set exists for the project.
+            # It's still worth checking the feature flag though, because we don't
+            # want to keep clustering projects for which the feature flag has been
+            # turned off.
```
Comment (jjbayer) [inline src/sentry/ingest/transaction_clusterer/tasks.py:66]:
```suggestion
```

Unrelated to this PR, but I noticed we do not check any feature flags anymore.

**Review by jjbayer — APPROVED** [review]

**Pull request merged**